src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -0,0 +1,103 @@
+package seedu.duke.manager;
+
+import seedu.duke.command.BadCommandException;
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Manager is the superclass for all WellNUS++ event drivers <br>
+ * <br>
+ * Each manager is in charge of 'managing' exactly one feature. <br>
+ * For example, <i>hb</i> and <i>reflect</i>. <br>
+ * <br>
+ * The manager should run an event driver (infinite loop) and is in charge
+ * of a Feature's input, output, 'business' logic and graceful termination
+ */
+public abstract class Manager {
+
+    private final CommandParser commandParser;
+    private ArrayList<Command> argumentList;
+
+    /**
+     * Construct a feature Manager to handle control flow for the given feature.
+     */
+    public Manager() {
+        this.commandParser = new CommandParser();
+    }
+
+    /**
+     * Utility function to get the CommandParser tied to the Manager class.
+     *
+     * @return commandParser reference to this manager's instance of CommandParser
+     */
+    public CommandParser getCommandParser() {
+        return this.commandParser;
+    }
+
+    /**
+     * Utility function to get the featureName this Manager is administering
+     *
+     * @return name of the feature that this Manager handles
+     */
+    public abstract String getFeatureName();
+
+    /**
+     * Utility function to get a summary description of the feature this Manager is administering
+     *
+     * @return summary description of the feature that this Manager handles
+     */
+    public abstract String getBriefDescription();
+
+    /**
+     * Utility function to get the full description of the feature this Manager is administering
+     *
+     * @return full description of the feature that this Manager handles
+     */
+    public abstract String getFullDescription();
+
+    /**
+     * Utility function to set a list of main arguments the feature supports
+     */
+    public abstract void setArgumentList();
```

Comment:
Hmm I feel that the argument list should be compile-time constant, so this shouldn't be necessary, or even desirable.

Reply:
The reason this setter is left in is to force all developers to properly populate the argument/command list. 

The idea is that this prevents developers from defining their own constants with their own names -- this will lead to integration issues with the methods that require a call to the arg/feature list, such as `isSupportedFeature`.

Unless we want all of these behaviours to be abstract